filters/builtin/decompress.go

```diff
@@ -0,0 +1,231 @@
+package builtin
+
+import (
+	"compress/flate"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+	"strings"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	// DecompressionNotPossible is the state-bag key to indicate
+	// to the subsequent filters during response processing that the
+	// content is compressed, but decompression was not possible, e.g
+	// because the encoding is not supported.
+	DecompressionNotPossible = "filter::decompress::impossible"
```

Comment:
nit: either call it DecompressionImpossible, or its value be something like `"filter::decompress::notpossible"` or `"filter::decompress::not-possible"`